reo/validators.py

```diff
@@ -1390,6 +1399,25 @@ def check_special_cases(self, object_name_path, template_values=None, real_value
                                             attr_name=key_name,
                                             time_steps_per_hour=ts_per_hour, number=number,
                                             input_isDict=input_isDict)
+            
+            if self.isValid:
+                if electric_tariff.get('coincident_peak_load_active_timesteps') is not None:
+                    for series in electric_tariff.get('coincident_peak_load_active_timesteps'):
+                        self.validate_timestep_series(series, 
+                        "ElectricTariff", 'coincident_peak_load_active_timesteps', 
+                        ts_per_hour, number=number, input_isDict=input_isDict)
+                    if len(electric_tariff.get('coincident_peak_load_active_timesteps')) != len(electric_tariff.get('coincident_peak_load_charge_us_dollars_per_kw')):
+                        self.input_data_errors.append(( "The number of rates in coincident_peak_load_charge_us_dollars_per_kw must"
+                                                        " match the number of timestep sets in coincident_peak_load_active_timesteps"))
+                    if self.isValid:
```

Comment:
is this second self.isValid necessary?

Reply:
@dylancutler yes, if both coincident peak inputs are not provided isValid will be false and in that case we don't want to run this additional check.

Reply:
@dylancutler yes, we don't want to run that default setting if the data is invalid